src/renderers/shared/utils/isTextInputElement.js

```diff
@@ -34,13 +34,11 @@ var supportedInputTypes: {[key: string]: true | void} = {
 };
 
 function isTextInputElement(elem: ?HTMLElement): boolean {
-  var nodeName = elem && elem.nodeName && elem.nodeName.toLowerCase();
-
-  if (nodeName === 'input') {
+  if (elem instanceof HTMLInputElement) {
```

Comment:
How much slower would it be to do something like:

```
getInstanceForNode(elem)._tag === 'input'
```

That gets us around working with the DOM.